### PR TITLE
feat: progress component

### PR DESCRIPTION
**Pull request by pauldambra**
We want a progress component...

This adds one using the HTML progress HTML component for horizontal progress bars
A custom component for an indeterminate circular progress (e.g. spinner)
And (faked for now) a custom component for a determinate circular progress bar (i.e. one with either val or max)

So, it let's you choose

* horizontal or circular
* spinner/indeterrminate or not
* val and max if determinate

I thought I'd stop here to check how close/far to/from expectations this lies

### CODE REVIEW

**Review by marandaneto — APPROVED** [review]
Left a [comment](https://github.com/PostHog/posthog/pull/19181/files#r1419140024) but its not a blocker to this PR, noice.

File: ee/frontend/mobile-replay/mobile.types.ts [hunk 1]
```diff
@@ -176,6 +176,19 @@ export type wireframeButton = wireframeInputBase & {
     value?: string
 }
 
+export type wireframeProgress = wireframeInputBase & {
+    inputType: 'progress'
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply ee/frontend/mobile-replay/mobile.types.ts:180]: Yeah, possibly....

I'll take a look at how much code would change in the transformer.

At the minute it's nice to have the types helping me but I bet I still coud

File: ee/frontend/mobile-replay/mobile.types.ts [hunk 2]
```diff
@@ -176,6 +176,19 @@ export type wireframeButton = wireframeInputBase & {
...
+    progressType: 'horizontal' | 'circular'
```
Comment (marandaneto) [inline ee/frontend/mobile-replay/mobile.types.ts:181]: should this be part of `style` instead? also fine to keep it here, but feels like part of it.

File: ee/frontend/mobile-replay/mobile.types.ts [hunk 3]
```diff
@@ -176,6 +176,21 @@ export type wireframeButton = wireframeInputBase & {
     value?: string
 }
 
+export type wireframeProgress = wireframeInputBase & {
+    inputType: 'progress'
+    /**
+     * @description This attribute specifies how much of the task that has been completed. It must be a valid floating point number between 0 and max, or between 0 and 1 if max is omitted. If there is no value attribute, the progress bar is indeterminate; this indicates that an activity is ongoing with no indication of how long it is expected to take.
+     */
+    value?: number
```
Comment (marandaneto) [inline ee/frontend/mobile-replay/mobile.types.ts:184]:
```suggestion
    progress?: number
```

That aligns better with the Android and iOS naming, but no strong opinions.
Is this the % of the `max`? eg 0.5 is 50% of the `max`? or is the current progress of the total?

Comment (pauldambra) [reply ee/frontend/mobile-replay/mobile.types.ts:184]: so... in html if no max is specified then it is assumed to be 1. e.g. 0.5 would be 50%

But if you specified a max of 10 then o.5 is 5%

TLDR; it's a value relative to the max

happy to switch from value to progress

Comment (pauldambra) [reply ee/frontend/mobile-replay/mobile.types.ts:190]: Actually I'm going to leave it as value, since that keeps `value` having the same meaning across multiple components
